workos/events.py

```diff
@@ -145,15 +150,8 @@ async def list_events(
             token=workos.api_key,
         )
 
-        if "default_limit" in locals():
-            if "metadata" in response and "params" in response["metadata"]:
-                response["metadata"]["params"]["default_limit"] = default_limit
-            else:
-                response["metadata"] = {"params": {"default_limit": default_limit}}
-
-        response["metadata"] = {
-            "params": params,
-            "method": AsyncEvents.list_events,
-        }
-
-        return response
+        return WorkOsListResource(
+            list_method=self.list_events,
+            list_args=params,
+            **ListPage[Event](**response).model_dump(),
+        )
```

Comment:
Oh interesting. Does this work without issue for async? I may have to revisit this in my PR.

Reply:
I doubt it. We probably need to parametrize the list resource to use an async client. But I haven't gotten there yet